openssl/src/cms.rs

```diff
@@ -219,11 +219,61 @@ impl CmsContentInfo {
             Ok(CmsContentInfo::from_ptr(cms))
         }
     }
+
+    /// Verify this CmsContentInfo's signature, given a stack of certificates
+    /// in certs, an X509 store in store. If the signature is detached, the
+    /// data can be passed in data. The data sans signature will be copied
+    /// into output_data if it is present.
+    ///
+    /// OpenSSL documentation at [`CMS_verify`]
+    ///
+    /// [`CMS_verify`]: https://www.openssl.org/docs/manmaster/man3/CMS_verify.html
+    pub fn verify(
+        &mut self,
+        certs: Option<&Stack<X509>>,
+        store: Option<&X509Store>,
+        data: Option<&[u8]>,
+        output_data: Option<&mut Vec<u8>>,
+        flags: CMSOptions,
+    ) -> Result<bool, ErrorStack> {
+        unsafe {
+            let certs = match certs {
+                Some(certs) => certs.as_ptr(),
+                None => ptr::null_mut(),
+            };
+            let store = match store {
+                Some(store) => store.as_ptr(),
+                None => ptr::null_mut(),
+            };
+            let in_data = match data {
+                Some(data) => MemBioSlice::new(data)?.as_ptr(),
```

Comment:
The MemBioSlice will be immediately freed, leaving in_data a dangling pointer.

Reply:
Implementation changed and a test added.